File: kotlin/0058-e-length-of-last-word.kt
// https://leetcode.com/problems/length-of-last-word/
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var count = 0
        var isFirstEmpty = true

        for (i in s.length -1 downTo 0) {
            if (s[i] != ' ' && isFirstEmpty) {
                isFirstEmpty = false
            }
            if (!isFirstEmpty) {
                if (s[i] == ' ') return count
                count++
            }
        }

        return count
    }
}

File: python/0058-e-length-of-last-word.py
# https://leetcode.com/problems/length-of-last-word/
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        count = 0
        clean = s.strip()
        for i in range(len(clean) - 1, -1, -1):
            if clean[i] == " ":
                return count
            count += 1
        
        return count